app/src/main/java/ar/com/wolox/android/training/ui/training/login/LoginPresenter.java

```diff
@@ -70,4 +90,44 @@ public void onSingUpButtonClicked() {
     public void onTermsAndConditionClicked() {
         getView().showTermsAndConditionWebView();
     }
+
+    private void sendLoginRequest(final String username, final String password, final Context ctx) {
+
+        Map<String, String> headerMap = new HashMap<>();
+        headerMap.put("Content-Type", mapKey);
+        headerMap.put("Accept", mapKey);
+
+        Call response = retrofitServices.getService(IRest.class).getUserRequest(headerMap, username, password);
+        response.enqueue(new Callback() {
+            @Override
+            public void onResponse(Call call, Response response) {
+                if (response.isSuccessful()) {
+                    try {
+                        if (response.body() != null) {
+                            JSONArray jsonArray = new JSONArray(response.body().toString());
+                            if (jsonArray.length() < 1) {
+                                getView().showCredentialsError(ERROR_CODE, ctx.getString(R.string.error_wrong_credentials));
```

Comment:
Since the presenter couldn't know nothing about the UI, that string should be setted from the view. Then, you won't need the context.